editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/BeanConfigUtil.java

```diff
@@ -0,0 +1,140 @@
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.search.IJavaSearchConstants;
+import org.eclipse.jdt.core.search.IJavaSearchScope;
+import org.eclipse.jdt.core.search.SearchEngine;
+import org.eclipse.jdt.core.search.SearchParticipant;
+import org.eclipse.jdt.core.search.SearchPattern;
+import org.eclipse.jdt.ui.IJavaElementSearchConstants;
+import org.eclipse.jdt.ui.JavaUI;
+import org.eclipse.jface.viewers.LabelProvider;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.dialogs.ElementListSelectionDialog;
+import org.eclipse.ui.dialogs.SelectionDialog;
+import org.fusesource.ide.camel.editor.internal.CamelEditorUIActivator;
+import org.fusesource.ide.camel.editor.utils.CamelUtils;
+import org.fusesource.ide.camel.model.service.core.catalog.Parameter;
+
+/**
+ * @author brianf
+ *
+ */
+public class BeanConfigUtil {
+	
+	private BeanConfigUtil() {
+		// private constructor
+	}
+
+	public static String handleClassBrowse(Shell shell) {
+        IJavaSearchScope scope = null;
+        IProject project = CamelUtils.project();
+        if (project != null) {
+        	IJavaProject jproject = JavaCore.create(project);
+	        if (jproject == null) {
+	            scope = SearchEngine.createWorkspaceScope();
+	        } else {
+                scope = SearchEngine.createJavaSearchScope(new IJavaElement[] { jproject });
+            }
+        }
+
+        try {
+            SelectionDialog dialog = JavaUI.createTypeDialog(shell, null, scope,
+                    IJavaElementSearchConstants.CONSIDER_CLASSES, false, "*Bean"); //$NON-NLS-1$
+            if (dialog.open() == SelectionDialog.OK) {
+                Object[] result = dialog.getResult();
+                if (result.length > 0 && result[0] instanceof IType) {
+                    return ((IType) result[0]).getFullyQualifiedName();
+                }
+            }
+        } catch (JavaModelException e) {
+			CamelEditorUIActivator.pluginLog().logError(e);
+        }
+        return null;
+    }
+
+	private static String openMethodDialog(IJavaProject jproject, String className, Shell shell) throws JavaModelException {
+		IType foundClass = jproject.findType(className);
+		if (foundClass != null) {
+			IMethod[] methods = foundClass.getMethods();
+			ElementListSelectionDialog dialog = new ElementListSelectionDialog(shell, new LabelProvider());
+			dialog.setTitle("Method Selection");
+			dialog.setMessage("Select a Method:");
+			dialog.setElements(methods);
+			if (dialog.open()  == SelectionDialog.OK) {
+				IMethod result = (IMethod) dialog.getFirstResult();
+				return result.getElementName();
+			}
+		}
+		return null;
+	}
+	
+    public static String handleMethodBrowse(String className, Shell shell) {
```

Comment:
would be better to provide the IProject as a parameter. it avoids to be dependent on the Workspace state